src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle'| 'circle' | 'rectangle';
+type Color = 'red'| 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || Math.max(a, b, c) >= (a + b + c) - Math.max(a, b, c)
+    ) {
+      throw new Error('it is not a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
 
+    return +(Math.sqrt(semiPerimeter
```

Comment:
Using `Number()` constuctor is more obvious
```suggestion
    return Number(Math.sqrt(semiPerimeter
```